From 898cec85907f8f171264c8be281a9ac2979b1655 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Schambacher?=
 <j-schambacher@users.noreply.github.com>
Date: Thu, 23 Jan 2020 13:32:13 +0100
Subject: [PATCH] Fix master mode settings of HiFiBerry DAC+ADC PRO
 card (#3424)

This patch fixes the board DAI setting when in master-mode.
Wrong setting could have caused random pop noise.

Signed-off-by: Joerg Schambacher <joerg@i2audio.com>
---
 sound/soc/bcm/hifiberry_dacplusadcpro.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/sound/soc/bcm/hifiberry_dacplusadcpro.c
+++ b/sound/soc/bcm/hifiberry_dacplusadcpro.c
@@ -285,6 +285,8 @@ static int snd_rpi_hifiberry_dacplusadcp
 
 		dai->name = "HiFiBerry DAC+ADC Pro";
 		dai->stream_name = "HiFiBerry DAC+ADC Pro HiFi";
+		dai->dai_fmt = SND_SOC_DAIFMT_I2S | SND_SOC_DAIFMT_NB_NF
+			| SND_SOC_DAIFMT_CBM_CFM;
 
 		// set DAC DAI configuration
 		ret = snd_soc_dai_set_fmt(rtd->codec_dais[0],
